### PR TITLE
Support missing namespace in parameter names in design_kw

### PR DESCRIPTION
**Pull request by berland**
First attempt at solving #59 

Check first if the behaviour imposed by the added tests is what we want, then assess whether the code achieving it was sufficiently elegant (perhaps it was not..)

### CODE REVIEW

**Comment by ertomatic** [comment]
Can one of the admins verify this patch?

File: tests/jobs/design_kw/test_design_kw.py [hunk 1]
```diff
@@ -78,6 +78,69 @@ def test_extract_key_value_dup_key():
...
+    "str_or_dict, expected",
+    [
+        ("foo", "foo"),
+        ("foo:bar", "bar"),
+        ("foo:bar:com", "bar:com"),
+        ("", ""),
+        ({"foo:bar": "value"}, {"bar": "value"}),
+        ({"foo:bar": "value", "foo:com": "value2"}, {"bar": "value", "com": "value2"}),
+        ({}, {}),
+        # Duplicates are dropped, a warning is logged.
```
Comment (markusdregi) [inline tests/jobs/design_kw/test_design_kw.py:91]: I don't think any warning is logged here...

File: semeio/jobs/design_kw/design_kw.py [hunk 2]
```diff
@@ -116,3 +119,53 @@ def extract_key_value(parameters):
     if errors:
         raise ValueError("\n".join(errors))
     return res
+
+
+def rm_genkw_prefix(str_or_dict, ignoreprefixes="LOG10_"):
```
Comment (markusdregi) [inline semeio/jobs/design_kw/design_kw.py:124]: Unless I've misunderstood the type of `str_or_dict` will always be a dict on the first call, and then the string support is used internally on the keys afterwards. I think the implementation would be much cleaner if we utilised this assumption and hid the string-support within the function such that `rm_genkw_prefix` only accepts dicts of depth one.

**Review by markusdregi — COMMENTED** [review]
I think this is a fair approach and the tests seems adequate 👍 I would like to see an added test for logging if that is the expected behaviour and in addition I think the implementation of `rm_genkw_prefix` could be simplified a bit.

I'm supporting this direction

File: tests/jobs/design_kw/test_design_kw.py (hunk 1 above)
Comment (berland) [reply tests/jobs/design_kw/test_design_kw.py]: Added extra duplicates warning.

File: semeio/jobs/design_kw/design_kw.py (hunk 2 above)
Comment (berland) [reply semeio/jobs/design_kw/design_kw.py]: New  implementation written - it got a lot better with only dict as input.

File: semeio/jobs/design_kw/design_kw.py [hunk 3]
```diff
@@ -116,3 +119,55 @@ def extract_key_value(parameters):
     if errors:
         raise ValueError("\n".join(errors))
     return res
+
+
+def rm_genkw_prefix(paramsdict, ignoreprefixes="LOG10_"):
+    """Remove anything before the first colon in a string, or remove
```
Comment (markusdregi) [inline semeio/jobs/design_kw/design_kw.py:125]: The doc string still need an update, besides that it looks good 👍 Nice clean up of `rm_genkw_prefix` ;)

Comment (berland) [reply semeio/jobs/design_kw/design_kw.py]: Updated.

**Review by markusdregi — APPROVED** [review]
🚀

**Pull request merged**